src/resolve/resolved-varlink.c

```diff
@@ -531,6 +555,58 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int vl_method_start_browse(Varlink* link, JsonVariant* parameters, VarlinkMethodFlags flags, void* userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                { "domain_name", JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, domain_name), JSON_MANDATORY },
+                { "name",        JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, name),        JSON_MANDATORY },
+                { "type",        JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, type),        JSON_MANDATORY },
+                { "ifname",      JSON_VARIANT_STRING,   json_dispatch_string, offsetof(LookupParametersMdnsBrowse, ifname),      JSON_MANDATORY },
+                { "token",       JSON_VARIANT_UNSIGNED, json_dispatch_uint64, offsetof(LookupParametersMdnsBrowse, token),       JSON_MANDATORY },
+                {}
+        };
+
+        _cleanup_(lookup_parameters_mdns_destroy) LookupParametersMdnsBrowse p = {};
+        Manager *m;
+        int r = 0;
+
+        assert(link);
+
+        m = varlink_server_get_userdata(varlink_get_server(link));
+        assert(m);
+
+        r = json_dispatch(parameters, dispatch_table, NULL, 0, &p);
+        if (r < 0) {
+                log_error_errno(r, "vl_method_start_browse json_dispatch fail. %m\n");
+                return r;
```

Comment:
log_error_errno() already returns the error you pass in anyway, hence write:

```c
return log_error_errno(…);
```

Also, the strings you pass to our logging functions will be suffixed with \n implicitly. Do not specify this manually.

and we usually write errors like this "Foobar failed: %m". i.e. use a colon as separator